debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputReplicationMessage.java

```diff
@@ -0,0 +1,318 @@
+            case "boolean":
+            case "bool":
+                return "t".equalsIgnoreCase(rawValue);
+
+            case "hstore":
+                return rawValue;
+
+            case "integer":
+            case "int":
+            case "int4":
+            case "smallint":
+            case "int2":
+            case "smallserial":
+            case "serial":
+            case "serial2":
+            case "serial4":
+            case "oid":
+                return Integer.valueOf(rawValue);
+
+            case "bigint":
+            case "bigserial":
+            case "int8":
+                return Long.valueOf(rawValue);
+
+            case "real":
+            case "float4":
+                return Float.valueOf(rawValue);
+
+            case "double precision":
+            case "float8":
+                return Double.valueOf(rawValue);
+
+            case "numeric":
+            case "decimal":
+                if ("NaN".equals(rawValue)) {
+                    return SpecialValueDecimal.NOT_A_NUMBER;
+                }
+                else {
+                    return new SpecialValueDecimal(new BigDecimal(rawValue));
+                }
+
+            case "character":
+            case "char":
+            case "character varying":
+            case "varchar":
+            case "bpchar":
+            case "text":
+                return rawValue;
+
+            case "date":
+                return DateTimeFormat.get().date(rawValue);
+
+            case "timestamp with time zone":
+            case "timestamptz":
+                return DateTimeFormat.get().timestampWithTimeZone(rawValue);
+
+            case "timestamp":
+            case "timestamp without time zone":
+                final LocalDateTime serverLocal = Conversions.fromNanosToLocalDateTimeUTC(DateTimeFormat.get().timestamp(rawValue));
+                return Conversions.toEpochNanos(serverLocal.toInstant(ZoneOffset.UTC));
+
+            case "time":
+                return rawValue;
+
+            case "time without time zone":
+                return DateTimeFormat.get().time(rawValue);
+
+            case "time with time zone":
+            case "timetz":
+                return DateTimeFormat.get().timeWithTimeZone(rawValue);
+
+            case "bytea":
+                // The string gets prefixed with '\xHEX_SEQUENCE"
+                return Strings.hexStringToByteArray(rawValue.substring(2));
+
+            case "box":
+                try {
+                    return new PGbox(rawValue);
+                } catch (final SQLException e) {
+                    LOGGER.error("Failed to parse point {}, {}", rawValue, e);
+                    throw new ConnectException(e);
+                }
+            case "circle":
+                try {
+                    return new PGcircle(rawValue);
+                } catch (final SQLException e) {
+                    LOGGER.error("Failed to parse circle {}, {}", rawValue, e);
```

Comment:
Let's either log or throw, but not both. Probably the latter here; the new exception can still have an enriched message.

Reply:
Fixed.